### PR TITLE
Override `protocArtifact` to version with M1 support

### PR DESCRIPTION
**Pull request by erichaagdev**
While working on #504, I ran into an issue building `build-caching-maven-samples` locally due to the `protoc` version bundled with `com.github.os72:protoc-jar-maven-plugin:3.11.4` not having M1 support.

image

Unfortunately, this appears to be the latest version of the plugin. However, from [the plugin's README](https://github.com/os72/protoc-jar-maven-plugin/blob/master/README.md?plain=1#L99), you can configure the plugin to use a non-bundled version of `protoc` by specifying the `protocArtifact` configuration property. This allows us to use a newer version of `protoc` which has M1 support.

I'm not sure if this is something we actually want to do, but it was easy enough to open a PR for so I figured I'd open one.

Note: The version I have configured is not the latest. It's just the first version with M1 support. 

Something else to consider is defining the `protoc` version as a property and referencing it for both `protocArtifact` and for the `protoc` project dependency.

### CODE REVIEW

**Review by alextu — APPROVED** [review]
Tried locally on non M1 Mac and it works fine

File: build-caching-maven-samples/protobuf-project/pom.xml [hunk 1]
```diff
@@ -32,6 +32,7 @@
             </goals>
             <configuration>
               <outputDirectory>target/generated-sources/protobuf</outputDirectory>
+              <protocArtifact>com.google.protobuf:protoc:3.17.3</protocArtifact>
```
Comment (jprinet) [inline build-caching-maven-samples/protobuf-project/pom.xml:35]: Maybe worth commenting that this is required for M1 support

**Review by jprinet — APPROVED** [review]
LGTM as this addressing M1 and still valid for non M1 environment

I would just comment the change as people will probably copy/paste and might end up with non strictly required configuration

File: build-caching-maven-samples/protobuf-project/pom.xml (hunk 1 above)
Comment (erichaagdev) [reply build-caching-maven-samples/protobuf-project/pom.xml:37]: Added in e978c05e9a080036d02ff34d4d3f2d302a76ce20

**Pull request merged**